courses/views.py

```
@@ -0,0 +1,122 @@
+        if self.request.user:
+            CourseTeacher.objects.create(course=saved_course, teacher=self.request.user)
+        messages.success(self.request, "Course saved successfully!")
+        return redirect(reverse("course_details", kwargs={"slug": saved_course.slug}))
+
+    def form_invalid(self, form):
+        print("Form errors:", form.errors)  # Log the errors to the console
+        messages.error(self.request, f"There was an error processing the form: {form.errors}")
+        return self.render_to_response(self.get_context_data(form=form))
+
+
+class CourseUpdateView(LoginRequiredMixin, UpdateView):
```

Ruff (0.8.2)

Undefined name `LoginRequiredMixin`

(F821)